OpenSim/Actuators/Millard2012EquilibriumMuscle.cpp

```diff
@@ -156,6 +156,12 @@ void Millard2012EquilibriumMuscle::extendFinalizeFromProperties()
     penMdl.set_optimal_fiber_length(getOptimalFiberLength());
     penMdl.set_pennation_angle_at_optimal(getPennationAngleAtOptimalFiberLength());
     penMdl.set_maximum_pennation_angle(get_maximum_pennation_angle());
+    try {
+        penMdl.finalizeFromProperties();
+    } catch (const SimTK::Exception::Base& ex) {
```

Comment:
ideally would only trap just the `InvalidPropertyValue` exception.

Reply:
The pennation model throws `SimTK::Exception::ErrorCheck` and `SimTK::Exception::ValueOutOfRange`, so catching `InvalidPropertyValue` won't fix #2106.

Reply:
I'll catch internally, but the code is a bit ugly: I need to pass the name of the property when throwing `InvalidPropertyValue`, so each property needs to be in a separate try/catch.

Reply:
Then don't do it. It would be cleaner/better to throw `InvalidPropertyValue` in the `MuscleFixedWidthPennationModel::extendFinalizeFromProperties()` (and in all subcomponents in general) regarding invalid property values.